src/Controller/Frontend/SearchController.php

```diff
@@ -37,13 +39,24 @@ public function __construct(Config $config, Environment $twig, TemplateChooser $
     public function search(ContentRepository $contentRepository, Request $request): Response
     {
         $page = (int) $request->query->get('page', 1);
+        $searchTerm = $request->get('searchTerm', $request->get('search', $request->get('q', '')));
         $amountPerPage = $this->config->get('general/listing_records');
 
         // @todo implement actual Search Engine
-        $records = $contentRepository->findForListing($page, $amountPerPage);
+        if (! empty($searchTerm)) {
+            $records = $contentRepository->searchNaive($searchTerm, $page, $amountPerPage);
+        } else {
+            $records = new Pagerfanta(new ArrayAdapter([]));
+        }
+
+        $context = [
+            'searchTerm' => $searchTerm,
+            'search' => $searchTerm, // Keep 'search' for Backwards Compatibility
```

Comment:
what breaks if you remove this line?

Reply:
Nothing in the default template. It'll break for people porting a Bolt 3 template, and/or people who are used to using `search`. It's a small effort to support it, and the impact is very low, but it will save some people some headaches.